### ISSUE TITLE
[a11y]: Pagination-JAWS does not announce the Expanded/Collapsed State of the page no ComboBox 

### ISSUE THREAD

**Issue by 2nikhiltom** [carbon-design-system/carbon#16567]
### Package

@carbon/react

### Package version

v1.56.0

### React version

18.02

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS

### Description

Expand/Collapse status for the page number box is not announced by screen reader 
<img width="899" alt="image" src="https://github.com/carbon-design-system/carbon/assets/63502271/8f3e5c86-bc72-4feb-9566-ff4224436293">
Screen reader expected to announce both Expand & Collapse status, make it  possible  by using ARIA Expand /Collapse properties accordingly.

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-pagination--default

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by guidari** [comment]
We are using the Select component that uses `select` tag from HTML. The `select` annouce when it's being closed and it announces the information when it's open.

Open State
![Image](https://github.com/user-attachments/assets/97ddcef6-61ed-4529-9b46-71dcf8085874)

Close state:
![Image](https://github.com/user-attachments/assets/35ff6cc1-3351-40ec-b2df-c198fc6471b3)
